### DISCUSSION TITLE
How to use aiogram + flask (or only aiogram) for payment processing in telegram bot?

### DISCUSSION THREAD

**Discussion by ubirust** [aiogram/aiogram#1128] [Q&A] [answered]
I have a telegram bot, it is written in python (uses the aiogram library), it works on a webhook. I need to process payments for a paid subscription to a bot (I use yoomoney as a payment).

It’s clear how you can do this on Flask: through its request method, catch http notifications that are sent from yoomoney (you can specify a url for notifications in yoomoney, where payment statuses like "payment.succeeded" should come)

In short, Flask is able to check the status of a payment. The bottom line is that the bot is written in aiogram and the bot is launched by the command:

`if __name__ == '__main__':
    try:
        start_webhook(
            dispatcher=dp,
            webhook_path=WEBHOOK_PATH,
            on_startup=on_startup,
            on_shutdown=on_shutdown,
            skip_updates=True,
            host=WEBAPP_HOST,
            port=WEBAPP_PORT
        )
    except (KeyboardInterrupt, SystemExit):
        logger.error("Bot stopped!")`

And if you just write in this code the launch of the application on flask in order to listen for answers from yoomoney, then EITHER the commands (of the bot itself) from aiogram will be executed OR the launch of flask, depending on what comes first in the code.

In fact, it is impossible to use flask and aiogram at the same time without multithreading. Is it possible somehow without flask in aiogram to track what comes to my server from another server (yoomoney)? Or how to use the aiogram + flask bundle more competently?

**Comment by mr-st0rm** [comment]
you can use aiohttp that user in aiogram for get requests from payment providers

**Comment by Olegt0rr** [comment, marked as answer]
`flask`- sync webserver
`aiohttp` - async webserver
`aiogram` - async framework based on `aiohttp`
`aiogram` and `flask` are not compatible

So, just use `aiohttp` instead

**Comment by vlnd0** [comment]
Hello, Mate. Faced with the same problem. How did you solve the problem of running aiohttp.web and aiogram polling together?

**Reply by Olegt0rr** [reply]
@vlnd0, what's the problem? 
[Here](https://github.com/Olegt0rr/WebServiceTemplate) you can find the example of using aiohttp.web.Application with aiogram polling / webhook (depends on settings)

P.S.: answer - bad place to asking questions
